components/mediators/bam/org.wso2.carbon.mediator.bam/src/main/java/org/wso2/carbon/mediator/bam/BamMediator.java

```diff
@@ -35,13 +38,48 @@
 
     private String serverProfile = "";
     private Stream stream = new Stream();
+    private boolean isContentAware = false;
+
+    public void setContentAwareness(StreamConfiguration streamConfiguration) {
+        List<StreamEntry> streamEntryList = streamConfiguration.getEntries();
+        for (StreamEntry streamEntry : streamEntryList) {
+            if ("$SOAPBody".equals(streamEntry.getValue())) { // Check if Dump Message Body was selected in UI
+                this.isContentAware = true;
+                return;
+            }
+        }
+        List<Property> properties = streamConfiguration.getProperties();
+        for (Property property : properties) {
+            if (property.isExpression() && property.getValue() != null) {
+                this.isContentAware = checkContentAware(property.getValue());
+            }
+        }
+    }
+
+    private boolean checkContentAware(String xpathString) {
+
+        boolean contentAware = false;
+
+        // For message body and xpath expressions on message body
+        if (xpathString.contains("/") || xpathString.contains("$body")) {
+            contentAware = true;
+        }
+
+        // For special property access
+        if (xpathString.contains("get-property('From'") ||
+            xpathString.contains("get-property('FAULT')")) {
+            contentAware = true;
+        }
+
+        return contentAware;
+    }
 
```

Comment:
Let's simplify this logic to prevent extra checks.

private boolean checkContentAware(String xpathString) {

        // For message body and xpath expressions on message body
        if (xpathString.contains("/") || xpathString.contains("$body")) {
            return true;
        }

        // For special property access
        if (xpathString.contains("get-property('From'") ||
            xpathString.contains("get-property('FAULT')")) {
            return true;
        }

        return false;
    }